src/cargo/ops/cargo_update.rs

```diff
@@ -787,45 +787,43 @@ fn report_latest(possibilities: &[IndexSummary], change: &PackageChange) -> Opti
             }
         })
         .filter(|s| package_id.version() != s.version() && version_req.matches(s.version()))
-        .max_by_key(|s| s.version())
-    {
+        .max_by_key(|s| s.version());
+    if let Some(summary) = compat_ver_compat_msrv_summary {
         let warn = style::WARN;
         let version = summary.version();
         let report = format!(" {warn}(available: v{version}){warn:#}");
         return Some(report);
     }
 
-    if let Some(summary) = possibilities
-        .iter()
-        .map(|s| s.as_summary())
-        .filter(|s| {
-            if let (Some(summary_rust_version), Some(required_rust_version)) =
-                (s.rust_version(), required_rust_version)
-            {
-                summary_rust_version.is_compatible_with(required_rust_version)
-            } else {
-                true
-            }
-        })
-        .filter(|s| is_latest(s.version(), package_id.version()))
-        .max_by_key(|s| s.version())
-    {
-        let warn = if change.is_transitive.unwrap_or(true) {
-            Default::default()
-        } else {
-            style::WARN
-        };
-        let version = summary.version();
-        let report = format!(" {warn}(available: v{version}){warn:#}");
-        return Some(report);
+    if !change.is_transitive.unwrap_or(true) {
```

Comment:
I might start forgetting things again. What does `is_transitive` mean here? It doesn't seem to be "transtive dependency" comparing to direct dependency. Or, is everything in `PackageDiff` is transitive?

Reply:
It is transitive vs direct dependency.

Reply:
Understood now. I missed this part that sets it back to false.

https://github.com/rust-lang/cargo/blob/9d5c1497f61d0e9a62e81c6f275e4750f404c420/src/cargo/ops/cargo_update.rs#L989-L994